pkg/session/token_store.go

```diff
@@ -0,0 +1,120 @@
+package session
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/percona/everest/pkg/common"
+	"github.com/percona/everest/pkg/kubernetes"
+)
+
+const (
+	sep          = ","
+	timestampLen = 10
+)
+
+func newTokenStore(ctx context.Context, kubeClient kubernetes.KubernetesConnector, logger *zap.SugaredLogger) (TokenStore, error) {
+	store := &tokenStore{
+		l:          logger,
+		kubeClient: kubeClient,
+	}
+	_, err := kubeClient.GetSecret(ctx, types.NamespacedName{Namespace: common.SystemNamespace, Name: common.EverestBlocklistSecretName})
+	if err == nil {
+		return store, nil
+	}
+	if !k8serrors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get secret: %w", err)
+	}
+	var createErr error
+	_, createErr = kubeClient.CreateSecret(ctx, blockListSecretTemplate(""))
+	if createErr != nil {
+		return nil, fmt.Errorf("failed to create secret: %w", createErr)
+	}
+	return store, nil
+}
+
+type tokenStore struct {
+	kubeClient kubernetes.KubernetesConnector
+	l          *zap.SugaredLogger
+	mutex      sync.RWMutex
+}
+
+func (ts *tokenStore) Add(ctx context.Context, shortenedToken string) error {
+	ts.mutex.Lock()
+	defer ts.mutex.Unlock()
```

Comment:
Do we actually need locking here (since we're not manually maintaining a cache anymore)?

The Kube API provides atomic updates to resources using resource versioning, so we already have protection from concurrent updates.. And we also already have retries on the caller side

Reply:
right, seems like the mutex not needed anymore. The resouce version protects from concurent updates and in case the update fails we have this retry mechanism. removed, thanks!